File: CPP/Rechtangle.cpp
#pragma once
#include"Rechtangle.h"
Rechtangle::Rechtangle(int32_t _x, int32_t _y, int32_t _height, int32_t _width) : x(_x), y(_y), height(_height), width(_width){}
Rechtangle::Rechtangle(const Rechtangle& r) 
{
	this->x = r.x;
	this->y = r.y;
	this->height = r.height;
	this->width = r.width;
}

int32_t Rechtangle::getX() const { return this->x; }
int32_t Rechtangle::getY()  const { return this->y; }
int32_t Rechtangle::getHeight() const { return this->height; }
int32_t Rechtangle::getWidth() const { return this->width; }
bool Rechtangle::operator < (const Rechtangle& right) const 
{
	if (this->x < right.x) return 1;
	else if (this->x > right.x) return 0;
	else if (this->y < right.y) return 1;
	else if (this->y > right.y) return 0;
	else return 1;
	/*if (this->width == right.width && this->height == right.height) return 0;
	else return 1;*/
}
void Rechtangle::setHeight(int32_t newHeight) { this->height = newHeight; }
void Rechtangle::setWidth(int32_t newWidth) { this->width = newWidth; }
void Rechtangle::setY(int32_t newY) { this->y = newY; }
void Rechtangle::setX(int32_t newX) { this->x = newX; }

File: JAVA/Rechtangle.java
package Picture;

public class Rechtangle {

    private int x = 0;
    private int y =0;
    private int height = 0;
    private int width = 0;

    public Rechtangle(int _x, int _y, int _height, int _width)
    {
        this.x=_x;
        this.y=_y;
        this.height=_height;
        this.width=_width;
    }

    public Rechtangle(Rechtangle r)
    {
        this.x=r.x;
        this.y=r.y;
        this.height=r.height;
        this.width= r.width;;
    }

    public int getHeight() {
        return height;
    }

    public int getWidth() {
        return width;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    boolean isLessThan (Rechtangle r)
    {
        if (this.x<r.x) return true;
        else if (this.x > r.x) return false;
        else if (this.y<r.y) return true;
        else if(this.y>r.y) return false;
        else return true;
    }
}
